content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,354 @@
+
+## Creating a new schema in the local database using migrations.
+
+Migrations help us build/alter our database schema in a consistent way. It uses the Ruby DSL (domain-specific language), and the dedicated rails ORM (Object-relational mapping) called Active Record so we don't have to write any SQL by hand.
+
+Our app requires a table and a column to store our todo items. We'll call that column `description`.
+To create this table with the column, we'll run `rails generate model Todo description`
+
+There are a few things to unpack about this:
+1. `rails generate`: compulsory prefix for generating various resources like models, controllers, etc.
+2. `model`: This tells rails that we intend to create a model. Creating a model will create a migration by default. Models represent the M in MVC and are part of the application that handles the business logic. How data is represented, accessed, and modified can be specified in the model. The model interacts with the database to carry out several operations.
+3. `Todo`: This is the name of the model we want to create. Rails will automatically pluralize this to create a table named `todos`.
+4. `description`: This is the column name we want our table to have. Columns are usually created with a default string datatype. We could do `clicks:integer` to add a column with an integer datatype instead.
+
+After running this command, you'll see a new file in the `db/migrate` folder. This file contains the migration code that will be used to create the table and the column in our database. The migration file name is in the format `yyyymmddhhmmss_create_todos.rb`. The first part of the file name is the timestamp of when the migration was created. This is used to keep track of the order in which migrations were created. The second part is the name is the kind of migration we want to create.
+
+Run `rails db:migrate` to apply the schema to our local database.
+
+A model file will also be created in the `app/models` folder. This file contains the model class and is used to define the model's attributes and relationships with other models.
+
+## Handling requests within controller methods.
+
+Controllers are responsible for handling requests and sending responses. They are the middleman between the user and the model. They receive requests from the user and send responses back to the user. They also interact with the model to get the data it needs to perform its operations.
+
+Let's create a controller for our todo items. Run `rails generate controller todos`. This will create a controller file in the `app/controllers` folder. It will also create a folder named `todos` in the `app/views` folder. This folder will contain all the views for our todo items.
```

Comment:
```suggestion
Let's create a controller for our todo items. In the terminal, run:

```shell
rails generate controller todos
```

This will create a controller file in the `app/controllers` folder. It will also create a folder named `todos` in the `app/views` folder. This folder will contain all the views for our todo items.
```

Reply:
@deyemiobaa For the main commands that will be run in the terminal, can we place them into shell blocks? This will make it easier to differentiate them from the rest of the bodies of explanatory text